### PR TITLE
Quaternion/update

### PR DESCRIPTION
**Pull request by samkys**
This PR is a Quaternion rotation from NED to ENU by doing a Quaternion multiplication. The previous method was to flip the x&z and invert z. The problem with the previous method is that when defining the TF tree in ROS there is a an underlying guess work to make sure the TF aligns properly. Using the `frame_based_enu` set to `true` you can now just define the TF to what is printed on the Vectornav itself.

Default parameters are exactly as they were so if a user doesn't want this option, they don't even need to worry about it.

### CODE REVIEW

**Comment by dawonn** [comment]
I'm confused by this PR, does this fix an error.in the prior logic? And why would you want to keep the old logic if it was broken? Or what new functionality does this provide? 

In general, I also like to do things the 'right way' (tm), so if this is that, then that's ok, I just want some clarification. Hope that makes sense.

File: src/main.cpp [hunk 1]
```diff
@@ -275,30 +281,57 @@ void BinaryAsyncMessageReceived(void* userData, Packet& p, size_t index)
         vec3f ar = cd.angularRate();
         vec3f al = cd.acceleration();
 
-        //Quaternion message comes in as a Yaw (z) Pitch (y) Roll (x) format
+        if (cd.hasAttitudeUncertainty())
+        {
+            vec3f orientationStdDev = cd.attitudeUncertainty();
+            msgIMU.orientation_covariance[0] = orientationStdDev[2]*orientationStdDev[2]*M_PI/180; // Convert to radians pitch
+            msgIMU.orientation_covariance[4] = orientationStdDev[1]*orientationStdDev[1]*M_PI/180; // Convert to radians Roll
+            msgIMU.orientation_covariance[8] = orientationStdDev[0]*orientationStdDev[0]*M_PI/180; // Convert to radians Yaw
```
Comment (dawonn-haval) [inline src/main.cpp]:
```suggestion
            msgIMU.orientation_covariance[8] = orientationStdDev[0]*orientationStdDev[0]*M_PI/180.0; // Convert to radians Yaw
```

Comment (dawonn-haval) [reply src/main.cpp]: etc... (yeah, it was bad before, I just hadn't noticed, Thanks!)

**Comment by samkys** [comment]
The only reason for keeping the old functionality was for backwards compatibility. If someone already defined their transform they could keep using it. Yes, ideally the old method would be completely removed, if you are ok with that I sure can. For anyone who was using the old way, the IMU TF defined in ROS would need to be changed to match what is printed on the Vectornav case (Which is the way it should be.....).

**Comment by samkys** [comment]
Hold off on this PR for just a little longer. I want to test it a bit more, submitted it just a little earlier than I should have.

**Comment by dawonn** [comment]
I need to think about this for a little bit...

It has been a while since I wrote it, but I thought there was a reason why I did not align the output to the triad engraved on the sensor. Something to do with the conversation to ENU made me think we couldn't keep it aligned with those markings.

So with your change, if you have the sensor working in ENU aligned with the markings, does that imply that the z axis points downward and the y axis points rightward if you mount it to the top of a ground vehicle?  The user would be able to just use a TF transformation to rotate the z to point up to get the the body frame, or mount the sensor upside down, perhaps?

**Comment by samkys** [comment]
No pressure, take your time. I think with the X&Y swap with Z inverted it doesn't allow for triad matching.

The way I have it mounted, Y points forward, X to the left and Z down. (You can mount this however you want now as the transform is just defined in your configuration file just like any other sensor.)

Attached are images of how the IMU is mounted on the actual robot and how the TF frame's are 
husky
image000000(1)
IMG_20190913_102904


visualized in RVIZ.

**Comment by samkys** [comment]
Just wanted to send a comment out. I have been using this quite a bit lately and it has been working fine. This is with a VN100.

**Pull request merged**